### PR TITLE
Reduce UI response data size for cluster overview page

### LINKED ISSUE #5929: Reduce amount of data transmitted for web-ui
The web-ui causes a fairly high amount of data to be transmitted. For example, if you have an average payload of 5MB every 5 seconds that leads to ~17k calls a day of 5MB each or ~86GB. One idea to mitigate this would be to reduce the data returned through the /ui/api/query endpoint to only include what is needed by the web-ui.

1. The query text in the UI looks to be truncated to ~250 characters, but the API return the full text.  So there % reduction here is unbounded.
2. The other data is session and stats.  Stats seem to be fully used, but for session only user name seems to be show, so I would guess about 50% reduction by replacing the session section with just user.

### PR DESCRIPTION
**Pull request by posulliv**
Resolves #5929

### CODE REVIEW

**Comment by findepi** [comment]
@dedep  @aalbu ptal

File: core/trino-main/src/main/java/io/trino/server/ui/UiBasicQueryInfo.java [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+        this(queryInfo.getQueryId(),
+                queryInfo.getSession().getUser(),
+                queryInfo.getSession().getPrincipal(),
+                queryInfo.getSession().getSource(),
+                queryInfo.getResourceGroupId(),
+                queryInfo.getState(),
+                queryInfo.getMemoryPool(),
+                queryInfo.isScheduled(),
+                queryInfo.getSelf(),
+                queryInfo.getQuery(),
```
Comment (aalbu) [inline core/trino-main/src/main/java/io/trino/server/ui/UiBasicQueryInfo.java:107]: Nit. Feels like query truncation fits better as a step in converting the `BasicQueryInfo`. The constructor could stay "dumb".

**Review by aalbu — APPROVED** [review]

**Comment by akshayrai** [comment]
@posulliv we have done something similar internally. Looking forward to have this change merged. Could you update and rebase the pull request! Thanks!

File: core/trino-main/src/main/java/io/trino/server/ui/UiQueryResource.java [hunk 2]
```diff
@@ -72,17 +72,17 @@ public UiQueryResource(DispatchManager dispatchManager, AccessControl accessCont
...
 
         List<BasicQueryInfo> queries = dispatchManager.getQueries();
         queries = filterQueries(extractAuthorizedIdentity(servletRequest, httpHeaders, alternateHeaderName, accessControl, groupProvider), queries, accessControl);
 
-        ImmutableList.Builder<BasicQueryInfo> builder = new ImmutableList.Builder<>();
+        ImmutableList.Builder<UiBasicQueryInfo> builder = new ImmutableList.Builder<>();
         for (BasicQueryInfo queryInfo : queries) {
             if (stateFilter == null || queryInfo.getState() == expectedState) {
-                builder.add(queryInfo);
+                builder.add(new UiBasicQueryInfo(queryInfo));
```
[the remark this answers is not in the document]
Comment (akshayrai) [reply core/trino-main/src/main/java/io/trino/server/ui/UiQueryResource.java:84]: @posulliv thanks! that makes sense. I think I misunderstood the api call with Query Detail.

**Comment by posulliv** [comment]
@findepi @aalbu do you think this is ready to be merged? Happy to do any further updates to get this merged.

**Comment by posulliv** [comment]
@findepi thank you for the many comments! I updated based on your review.

**Pull request merged**